pig/src/org/partiql/pig/domain/parser/TypeDomainParser.kt

```diff
@@ -15,275 +15,362 @@
 
 package org.partiql.pig.domain.parser
 
-import com.amazon.ionelement.api.*
 import com.amazon.ion.IonReader
 import com.amazon.ion.system.IonReaderBuilder
-import org.partiql.pig.domain.model.*
+import com.amazon.ionelement.api.AnyElement
+import com.amazon.ionelement.api.IonElement
+import com.amazon.ionelement.api.IonElementException
+import com.amazon.ionelement.api.IonElementLoaderOptions
+import com.amazon.ionelement.api.IonLocation
+import com.amazon.ionelement.api.MetaContainer
+import com.amazon.ionelement.api.SexpElement
+import com.amazon.ionelement.api.SymbolElement
+import com.amazon.ionelement.api.createIonElementLoader
+import com.amazon.ionelement.api.emptyMetaContainer
+import com.amazon.ionelement.api.head
+import com.amazon.ionelement.api.location
+import com.amazon.ionelement.api.metaContainerOf
+import com.amazon.ionelement.api.tag
+import com.amazon.ionelement.api.tail
+import org.partiql.pig.domain.model.Arity
+import org.partiql.pig.domain.model.DataType
+import org.partiql.pig.domain.model.NamedElement
+import org.partiql.pig.domain.model.PermutedDomain
+import org.partiql.pig.domain.model.PermutedSum
+import org.partiql.pig.domain.model.Statement
+import org.partiql.pig.domain.model.Transform
+import org.partiql.pig.domain.model.TupleType
+import org.partiql.pig.domain.model.TypeDomain
+import org.partiql.pig.domain.model.TypeRef
+import org.partiql.pig.domain.model.TypeUniverse
+import org.partiql.pig.errors.ErrorContext
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.InputStream
+import java.util.Stack
+
+/**
+ * Parses the type universe specified in [universeFilePath].
+ *
+ * Any files included with `include_file` are relative to the directory containing [universeFilePath].
+ */
+internal fun parseTypeUniverseFile(universeFilePath: String): TypeUniverse {
+    val parser = TypeUniverseParser(FILE_SYSTEM_INPUT_STREAM_SOURCE)
+    return parser.parseTypeUniverse(universeFilePath)
+}
+
+internal class TypeUniverseParser(
+    private val imporSource: InputStreamSource
```

Comment:
nit: could this be called `includeSource` for consistency with `include...`?

Reply:
Yeah, the typo made me not find this when I was searching for the word "import".  Thanks.